### PR TITLE
Add project sorting by name and last used state

### LINKED ISSUE #182: TimeTrackerComponent: Project AC: Show last used projects first

### PR DESCRIPTION
**Pull request by jakobneumann**
Resolves #182

### CODE REVIEW

**Comment by Alex-Witkowski** [comment]
Hey @gregor-doroschenko would be great if you do the review ;)

File: src/app/components/time-tracker/time-tracker.component.ts [hunk 1]
```diff
@@ -106,9 +106,41 @@ export class TimeTrackerComponent implements OnInit, OnChanges {
...
     }
   }
 
+  /**
+   * Sorts the projects by name alphabetically and moves recently used ones first
+   */
+  sortProjects() {
+    this.projects = this.projects.sort((p1,p2) => {
+      if (this.recentlyUsedIndex(p1) === Infinity && this.recentlyUsedIndex(p2) === Infinity) {
+        if (p1.name.toLowerCase() < p2.name.toLowerCase()) {
```
Comment (gregor-doroschenko) [inline src/app/components/time-tracker/time-tracker.component.ts]: Please use shorthand if-else

File: src/app/components/time-tracker/time-tracker.component.ts [hunk 2]
```diff
@@ -106,9 +106,41 @@ export class TimeTrackerComponent implements OnInit, OnChanges {
...
+          return -1;
+        }
+        return 1;
+      }
+      if (this.recentlyUsedIndex(p1) < this.recentlyUsedIndex(p2)) {
```
Comment (gregor-doroschenko) [inline src/app/components/time-tracker/time-tracker.component.ts]: Please use shorthand if-else

File: src/app/components/time-tracker/time-tracker.component.ts (hunk 1 above)
Comment (jakobneumann) [reply src/app/components/time-tracker/time-tracker.component.ts]: Changed by new commit

File: src/app/components/time-tracker/time-tracker.component.ts (hunk 2 above)
Comment (jakobneumann) [reply src/app/components/time-tracker/time-tracker.component.ts]: Changed by new commit

**Pull request merged**